### PR TITLE
[Tune] Remove `checkpoint_at_end` from tune+serve docs

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by amogkam** [comment]
@krfricke can you confirm that this example still works with `checkpoint_at_end` removed?

**Review by richardliaw — COMMENTED** [review]
lgtm, will merge after kai approves. thanks!

**Review by krfricke — APPROVED** [review]
Works!

**Pull request merged**
